fuse/eval/metrics/sequence_gen/metrics_seq_gen_common.py

```diff
@@ -18,13 +18,99 @@
 """
 from typing import Optional, Tuple, List
 from functools import partial
-
+from copy import copy
 import torch
 import numpy as np
 
 from fuse.eval.metrics.metrics_common import MetricPerBatchDefault
 
 
+class MetricCountSeqAndTokens(MetricPerBatchDefault):
+    """
+    Counts the total number sequences and tokens in encoder_input
+    """
+
+    def __init__(
+        self,
+        encoder_input: str,
+        ignore_index: Optional[int] = None,
+        state: Optional[dict] = None,
+        **kwargs: dict,
+    ) -> None:
+        """
+        :param encoder_input: key to the encoder_input
+        :param ignore_index: token_id to ignore (not to count), typically pad token id
```

Comment:
I think it should be able to support a list of token ids to ignore. Unless you want to enforce the user to ignore only the PAD one.

Reply:
I went with just one to be more efficient - typically we would like to just skip the padding.